src/core/license.rs

```diff
@@ -146,13 +149,12 @@ impl LicenseMessage {
             "securityFlagHi" => U16::LE(0)
         ];
         security_header.read(&mut stream)?;
-        if (cast!(DataType::U16, security_header["securityFlag"])?
-            & SecurityFlag::SecLicensePkt as u16)
-            == 0
-        {
+
+        let security_flag = cast!(DataType::U16, security_header["securityFlag"])?;
+        if (security_flag & SecurityFlag::SecLicensePkt as u16) == 0 {
             return Err(Error::RdpError(RdpError::new(
                 RdpErrorKind::InvalidData,
-                "SEC: Invalid Licence packet",
+                &format!("SEC: Invalid Licence packet (flag={security_flag:x})"),
```

Comment:
Unless the RDP protocol uses the british spelling...

```suggestion
                &format!("SEC: Invalid license packet (flag={security_flag:x})"),
```